src/fit_dataframe.jl

```diff
@@ -269,16 +268,12 @@ function expand_categoricals!(df::DataFrame,categoricals::Array)
     return expand_categoricals!(df, categoricalidxs)
 end
 
-# convert NaNs to NAs
-# isnan(x::NAtype) = false
-isnan(x::AbstractString) = false
```

Comment:
this was type-piracy.
Fixed the problem that it was avoiding via adding `isa Number` check below